org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -765,6 +765,17 @@ private boolean isEnablePreviewFeatures(MavenProject mavenProject, MojoExecution
     } catch(CoreException ex) {
       //ignore
     }
+
+    //3nd, check the --enable-preview flag in the ${maven.compiler.enablePreview}
+    try {
+      boolean enablePreview = maven.getMojoParameterValue(mavenProject, execution, "enablePreview", boolean.class, //$NON-NLS-1$
+          monitor);
+      if(enablePreview) { 
```

Comment:
We cannot use the primitive boolean here as this will always lead to a class-cast exception because internally Objects are used. But using Boolean solves this.
```suggestion
      Boolean enablePreview = maven.getMojoParameterValue(mavenProject, execution, "enablePreview", Boolean.class, //$NON-NLS-1$
          monitor);
      if(Boolean.TRUE.equals(enablePreview)) {
```